docs/content/docs/reference/react/protecting-routes.mdx

```diff
@@ -4,49 +4,67 @@ title: 'Protecting routes'
 
 ## React Router
 
-You can protect routes by creating an `AuthGate` component when using `@nhost/react` with [React Router](https://reactrouter.com/web/guides/quick-start).
+You can protect routes by creating a wrapper component, `ProtectedRoute`, to implement the authentication logic using `@nhost/react`.
+
+> This example uses the latest version of [React Router (v6)](https://reactrouter.com/docs/en/v6).
 
 ```jsx
-import { useAuthenticationStatus } from '@nhost/react'
-import { Redirect } from 'react-router-dom'
+// src/components/ProtectedRoute.js
+
+import { useAuthenticationStatus } from '@nhost/react';
+import { Navigate, useLocation } from 'react-router';
 
-export function AuthGate({ children }) {
-  const { isLoading, isAuthenticated } = useAuthenticationStatus()
+const ProtectedRoute = ({ children }) => {
+  const { isAuthenticated, isLoading } = useAuthenticationStatus();
+  const location = useLocation();
 
   if (isLoading) {
-    return <div>Loading...</div>
+    return (
+      <div>Loading...</div>
+    );
   }
 
   if (!isAuthenticated) {
-    return <Redirect to="/login" />
+    return <Navigate to="/login" state={{ from: location }} replace />;
```

Comment:
For the sake of demonstration I think we can keep it simple and just remove the `state` prop here

Reply:
I don't think it is adding much complexity. But you're right; let's remove it since it's the Nhost docs and not the React Router one 😉

Reply:
Precisely 😄  

Thanks!